ir/src/depgraph.rs

```diff
@@ -0,0 +1,433 @@
+use filetime::FileTime;
+use serde::{Deserialize, Serialize};
+
+use std::{
+    collections::{HashMap, HashSet},
+    fs, io,
+    path::{Path, PathBuf},
+    vec,
+};
+
+/// Helps identify changes in a set of files.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(from = "DepGraphSerdeRepr", into = "DepGraphSerdeRepr")]
+pub struct DepGraph {
+    entries: HashMap<PathBuf, Option<DepGraphEntry>>,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+struct DepGraphEntry {
+    mtime: FileTime,
+    size: u64,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct DepGraphSerdeRepr {
```

Comment:
so this is used entirely for serialization? If so, does it need to be public? If we can get away with it I might be tempted to just put all the serialization code in a separate module.

Reply:
I think the pub is a copy/paste error, fixing